modules/passkey/src/utils/webauthn.ts

```diff
@@ -0,0 +1,186 @@
+import { ethers } from 'ethers'
+import type { BigNumberish, BytesLike } from 'ethers'
+import CBOR from 'cbor'
+
+/**
+ * Returns the flag for the user verification requirement.
+ *
+ * See: <https://w3c.github.io/webauthn/#enumdef-userverificationrequirement>
+ */
+export function userVerificationFlag(userVerification: UserVerificationRequirement = 'preferred'): number {
+  switch (userVerification) {
+    case 'preferred':
+      return 0x01
+    case 'required':
+      return 0x04
+    default:
+      throw new Error(`user verification requirement ${userVerification} not supported`)
+  }
+}
+
+/**
+ * Returns the message that gets signed by the WebAuthn credentials.
+ *
+ * See <https://w3c.github.io/webauthn/#fig-signature>
+ */
+export function encodeWebAuthnSigningMessage(
+  clientData: { type: 'webauthn.get'; challenge: string; [key: string]: unknown },
+  authenticatorData: BytesLike,
+) {
+  return ethers.getBytes(ethers.concat([authenticatorData, ethers.sha256(ethers.toUtf8Bytes(JSON.stringify(clientData)))]))
+}
+
+/**
+ * Encode bytes using the Base64 URL encoding.
+ *
+ * See <https://www.rfc-editor.org/rfc/rfc4648#section-5>
+ *
+ * @param data data to encode to `base64url`
+ * @returns the `base64url` encoded data as a string.
+ */
+export function base64UrlEncode(data: BytesLike | ArrayBufferLike): string {
+  const bytes = ethers.isBytesLike(data) ? data : new Uint8Array(data)
+  return ethers.encodeBase64(bytes).replace(/\+/g, '-').replace(/\//g, '_').replace(/=*$/, '')
+}
```

Comment:
can be encoded without ethers:
https://github.com/safe-global/safe-modules/blob/89c46b2c9a27fec6ffbaefc6391742e6b76e41c3/modules/passkey/test/libraries/WebAuthn.spec.ts#L5-L11

Reply:
`Buffer` is not a browser API AFAIK.

Reply:
Right! I just found this, seems like there are browser's built-in functions: https://www.digitalocean.com/community/tutorials/how-to-encode-and-decode-strings-with-base64-in-javascript#encoding-and-decoding-strings-with-base64